File: mobile/lib/features/course/domain/entities/course.dart
import 'package:equatable/equatable.dart';

import 'course_image.dart';

class Course extends Equatable {
  final String id;
  final String name;
  final String description;
  final CourseImage image;
  final int numberOfChapters;
  final String departmentId;
  final String? referenceBook;
  final String ects;
  final int grade;
  final bool cariculumIsNew;

  const Course({
    required this.id,
    required this.name,
    required this.departmentId,
    required this.description,
    required this.numberOfChapters,
    required this.ects,
    required this.image,
    this.referenceBook,
    required this.grade,
    required this.cariculumIsNew,
  });

  @override
  List<Object?> get props => [
        id,
        name,
        description,
        numberOfChapters,
        departmentId,
        ects,
        image,
        grade
      ];
}

File: backend/src/models/course.ts
import { Document, Schema, model } from "mongoose";
export interface ICourse extends Document {
//   id: Schema.Types.ObjectId;
name: String;
image: Schema.Types.ObjectId;
description: String;
noOfChapters: Number;
grade: Number;
curriculum: Boolean;
departmentId: Schema.Types.ObjectId;
referenceBook: String;
ECTS: Number;
}

const CourseSchema: Schema<ICourse> = new Schema(
{
  name: {
    type: String,
    required: [true, "Course name is required"],
  },
  image: {
    type: Schema.Types.ObjectId,
    ref: "Image",
    default: null,
    required: false,
  },
  description: {
    type: String,
    required: false,
    default: "",
  },
  noOfChapters: {
    type: Number,
    required: false,
    default: 0,
  },
  grade: {
    type: Number,
    required: false,
    default: 0,
  },
  curriculum:{
    type: Boolean,
    required: true,
    default: true
  },
  departmentId: {
    type: Schema.Types.ObjectId,
    ref: "Department",
    required: [true, "Department Id is required"],
  },
  referenceBook: {
    type: String,
    required: false,
    default: "No reference book."
  },
  ECTS: {
      type: Number,
      default:0
  },
},
{
  timestamps: {
    createdAt: "createdAt",
    updatedAt: "updatedAt",
  },
}
);

const Course = model<ICourse>("Course", CourseSchema);

export default Course;
